packages/legend-application-studio/src/components/editor/edit-panel/service-editor/ServiceExecutionQueryEditor.tsx

```diff
@@ -247,16 +255,17 @@ export const ServiceExecutionQueryEditor = observer(
     isReadOnly: boolean;
   }) => {
     const { executionState, isReadOnly } = props;
-    const applicationStore = useApplicationStore();
+    const applicationStore = useLegendStudioApplicationStore();
     const editorStore = useEditorStore();
     const queryState = executionState.queryState;
     const embeddedQueryBuilderState = editorStore.embeddedQueryBuilderState;
+    const service = executionState.serviceEditorState.service;
     // actions
     const editWithQueryBuilder = (
-      openQueryBuilderTextEditor: boolean,
+      editQueryInFormMode: boolean,
```

Comment:
let's only have one flag here. 
`openInTextMode = false`. This should default to false.
This will be triggered under the 'text' option for edit query. 
for the double click, lets just open query builder as we have always done.